File: 300/383.ransom-note.py
def canConstruct(ransomNote: str, magazine: str) -> bool:
    if len(magazine) < len(ransomNote):
        return False

    r = list(ransomNote)
    m = list(magazine)
    for c in ransomNote:
        if c in m:
            m.remove(c)
            r.remove(c)

    return len(r) == 0

if __name__ == "__main__":
    ransomNote = "fffbfg"
    magazine = "effjfggbffjdgbjjhhdegh"

    print(canConstruct(ransomNote, magazine))

File: 300/383.ransom-note.go
package main

import "fmt"

func remove(s []rune, r rune) []rune {
	for i, v := range s {
		if v == r {
			return append(s[:i], s[i+1:]...)
		}
	}
	return s
}

func canConstruct(ransomNote string, magazine string) bool {
	rRunes := []rune(ransomNote)
	mRunes := []rune(magazine)
	target := []rune(ransomNote)

	for _, r := range rRunes {
		isContain := false
		for _, m := range mRunes {
			if r == m {
				isContain = true
			}
		}

		if isContain {
			mRunes = remove(mRunes, r)
			target = remove(target, r)
		}
	}
	return len(target) == 0
}

func main() {
	ransomNote := "aab"
	magazine := "baa"
	fmt.Println(canConstruct(ransomNote, magazine))
}
